### PR TITLE
ddl: make TestTableSplit stable

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve? 
```
FAIL: table_split_test.go:31: testDDLTableSplitSuite.TestTableSplit
table_split_test.go:54:
c.Assert(loc.StartKey, BytesEquals, []byte(regionStartKey))
... bytes_one []uint8 = []byte {0x74, 0x80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xd}
... bytes_two []uint8 = []byte {0x74, 0x80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xf}
```
The [LocateKey](https://github.com/pingcap/tidb/compare/master...zimulala:table_split?expand=1#diff-b7efc2112386df16fee3e3cd363a8203R54) gets the region from region cache.
But region cache may be out of date.

### What is changed and how it works?
We need to drop the expired region and load it again.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @disksing @tiancaiamao

**Comment by disksing** [comment]
LGTM

File: ddl/table_split_test.go [hunk 1]
```diff
@@ -51,5 +53,11 @@ func (s *testDDLTableSplitSuite) TestTableSplit(c *C) {
 	cache := store.(kvStore).GetRegionCache()
 	loc, err := cache.LocateKey(tikv.NewBackoffer(context.Background(), 5000), regionStartKey)
 	c.Assert(err, IsNil)
-	c.Assert(loc.StartKey, BytesEquals, []byte(regionStartKey))
+	if bytes.Compare(loc.StartKey, []byte(regionStartKey)) != 0 {
+		// Region cache may be out of date, so we need to drop this expired region and load it again.
+		cache.DropRegion(loc.Region)
+		loc, err := cache.LocateKey(tikv.NewBackoffer(context.Background(), 5000), regionStartKey)
```
Comment (tiancaiamao) [inline ddl/table_split_test.go]: What happens when the result is stale again, should it be a while condition rather than if condition?

Comment (zimulala) [reply ddl/table_split_test.go]: Why is it stale again?

Comment (tiancaiamao) [reply ddl/table_split_test.go]: > The LocateKey gets the region from region cache.
> But region cache may be out of date.

**Pull request merged**